tests/python/direct/test_narrow_precision.py

```diff
@@ -274,3 +275,35 @@ def nvfuser_fusion_id0(fd: FusionDefinition) -> None:
         )
 
     assert torch.allclose(o_decomposed_ref, o[0], atol=1e-2, rtol=1e-2)
+
+
+@pytest.mark.skipif(
+    is_pre_blackwell(), reason="Only supported on blackwell and newer devices."
+)
+@pytest.mark.skipif(
+    not microarchitecture_is_pre(12), reason="Does not support blackwell compute 12.0"
+)
+@pytest.mark.parametrize("dtype", [torch.bfloat16, torch.float])
+def test_fp4_vectorization(
+    nvfuser_direct_test,
+    dtype,
+):
+    inputs = [
+        torch.ones(4, 8, dtype=dtype, device="cuda"),
+        torch.ones(4, dtype=dtype, device="cuda"),
+    ]
+
+    def nvfuser_fusion_id0(fd: FusionDefinition) -> None:
+        T0 = fd.from_pytorch(inputs[0])
+        T1 = fd.from_pytorch(inputs[1])
+        T2 = fd.ops.cast(T0, DataType.Float)
+        cast_T1 = fd.ops.cast(T1, DataType.Float)
+        broadcast_T1 = fd.ops.broadcast(cast_T1, [False, True])
+        T3 = fd.ops.div(T2, broadcast_T1)
+        T4 = fd.ops.cast(T3, DataType.Float4_e2m1fn)
+        T5 = fd.ops.reshape(T4, [32])
+        fd.add_output(T5)
+
+    o, _ = nvfuser_direct_test.exec_nvfuser(nvfuser_fusion_id0, inputs)
+
+    ref_o = to_fp4(inputs[0].to(torch.float) / inputs[1].unsqueeze(-1)).reshape(-1)
```

Comment:
Can somebody teach me how validation is done here? IIUC, `exec_nvfuser` just executes the fusion. There's `ref_o`. Shouldn't we compare `o` and `ref_o`? 

Please don't use names like `o`. Single-letter names should be avoided except established common names like `i` for loop indices.

Reply:
And also, this is for vectorization, but I'm not seeing where the vectorization is validated.